packages/app-desktop/gui/NoteListControls/NoteListControls.tsx

```diff
@@ -69,6 +71,73 @@ const SortOrderButtonsContainer = styled.div`
 
 function NoteListControls(props: Props) {
 	const searchBarRef = useRef(null);
+	const newNoteRef = useRef(null);
+	const newTodoRef = useRef(null);
+	const noteControlsRef = useRef(null);
+	const searchAndSortRef = useRef(null);
+
+	const [noteBtnText, setNoteBtnText] = useState('New note');
+	const [todoBtnText, setTodoBtnText] = useState('New to-do');
+	const [breakpoint, setBreakpoint] = useState('l');
+	const [iconNote, setIconNote] = useState('fas fa-plus');
+	const [iconTodo, setIconTodo] = useState('fas fa-plus');
+
+	useEffect(() => {
+		const breakpoints = [{ s: 135 }, { md: 189 }, { l: 470 }, { xl: 500 }];
+		// Find largest breakpoint that width is less than
+		const index = breakpoints.map(x => Object.values(x)[0])
+			.findIndex(x => props.width < x);
+
+		const bp = index === -1 ? Object.keys(breakpoints[breakpoints.length - 1])[0] : Object.keys(breakpoints[index])[0];
+
+		setBreakpoint(bp);
+	}, [props.width]);
```

Comment:
To calculate derived values, it's better and more explicit to use `useMemo`, so you can have `const breakpoint = useMemo(....)` and it means you can remove state management for `breakpoint`

Reply:
I changed it. Let me know if this is what you had in mind.